### ISSUE TITLE
@dataclass named arguments have wrong type in hover text

### ISSUE THREAD

**Issue by falan** [microsoft/pyright#731]
**Describe the bug**
`@dataclass` have wrong  type hint on hover when data type is correct.

**To Reproduce**
```python
from dataclasses import dataclass

@dataclass
class DataclassData:
    a: str
    b: str

datatclassData = DataclassData(a=3, b="")
datatclassData = DataclassData(a="", b="")
```

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/4638204/84634030-d0cc7d80-af23-11ea-9277-1d0597fe8103.png)
![image](https://user-images.githubusercontent.com/4638204/84634179-08d3c080-af24-11ea-81b8-1e44a05dcaa0.png)

**VS Code extension or command-line**
VS Code extension 1.1.42
Python 3.7.4

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.43, which I just published.
